core/scripts/chaincli/handler/keeper_upkeep_events.go

```diff
@@ -63,3 +64,53 @@ func (k *Keeper) UpkeepCounterEvents(ctx context.Context, hexAddr string, fromBl
 		}
 	}
 }
+
+func (k *Keeper) UpkeepPokemonEvents(ctx context.Context, hexAddr string, fromBlock, toBlock uint64) {
+	contractAddress := common.HexToAddress(hexAddr)
+	upkeepCounter, err := upkeep_apifetch_wrapper.NewUpkeepAPIFetch(contractAddress, k.client)
+	if err != nil {
+		log.Fatalln("Failed to create a new upkeep pokemon", err)
+	}
+	filterOpts := bind.FilterOpts{
+		Start:   fromBlock,
+		End:     &toBlock,
+		Context: ctx,
+	}
+	upkeepIterator, err := upkeepCounter.FilterPokemonUpkeep(&filterOpts, nil)
```

Comment:
if the query takes too long, we can split the block range and batch them. but it seems filtering logs is very fast according to my Gauntlet experience.